iam/snippets/src/main/java/QueryTestablePermissions.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START iam_query_testable_permissions]
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.cloud.iam.admin.v1.IAMClient.QueryTestablePermissionsPagedResponse;
+import com.google.iam.admin.v1.QueryTestablePermissionsRequest;
+import java.io.IOException;
+
+public class QueryTestablePermissions {
+  public static void main(String[] args) {
+    // TODO(developer): Replace the variable before running the sample.
+    String projectId = "your-project-id";
+
+    queryTestablePermissions(projectId);
+  }
+
+  public static void queryTestablePermissions(String projectId) {
+    QueryTestablePermissionsRequest queryTestablePermissionsRequest =
+        QueryTestablePermissionsRequest.newBuilder()
+            .setFullResourceName("//cloudresourcemanager.googleapis.com/projects/" + projectId)
```

Comment:
Should the resource name be a variable passed into the method? 
Can we move it to the main method and ask the developer to update that variable based on their needs and give them an example?

Reply:
It would be nice to know where this value comes from and if it is changeable.